csrc/runtime/fusion_kernel_runtime.cpp

```diff
@@ -196,6 +199,28 @@ flatbuffers::Offset<serde::FusionKernelRuntime> FusionKernelRuntime::serialize(
       segmented_fusion_fb);
 }
 
+namespace {
+std::vector<Expr*> toposortExprs(
+    SegmentedFusion* fusion,
+    SegmentedGroup* group) {
+  std::vector<Expr*> sorted_exprs;
+  {
+    auto
+        [/*IrCloner*/ group_ir_cloner,
+         /*std::unique_ptr<Fusion>*/ group_fusion] = fusion->makeFusion(group);
```

Comment:
makeFusion is unfortunately more expensive than you might think. It copies the **complete** fusion rather than just Exprs in the group, which is unnecessary but has been there for legacy reasons. 

Did you try StmtSort::getExprsBetween or similar helpers?